### PR TITLE
Modified the changes according to RD-2714 of sdk-java

### PR DESCRIPTION
**Pull request by priyanshu-ap**

### CODE REVIEW

File: tpp/src/http/AuthHttpClient.js [hunk 1]
```diff
@@ -401,6 +406,50 @@ class AuthHttpClient extends CoreAuthHttpClient {
...
+    /**
+     * Get url to bank authorization page for a token request.
+     *
+     * @param bankId {string} Bank Id
+     * @param tokenRequestId {string} Token request id
+     * @returns {Object} response to the api call
+     */
+    async getBankAuthUrl(bankId, tokenRequestId){
+        const request = {
+            method: 'post',
```
Comment (daniellin215) [inline tpp/src/http/AuthHttpClient.js]: is this a 'GET' call?

File: core/src/http/AuthHeader.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ class AuthHeader {
             'created-at-ms=' + now +
             AuthHeader._onBehalfOfHeader(context) +
             AuthHeader._customerInitiated(context);
-
+
```
Comment (yuhanniu) [inline core/src/http/AuthHeader.js]: nit: revert

File: core/src/http/AuthHttpClient.js [hunk 3]
```diff
@@ -646,12 +619,12 @@ export class AuthHttpClient {
...
         const versionHeader = new VersionHeader();
         const developerHeader = new DeveloperHeader(this._developerKey);
-        const securityMetadataHeader = new SecurityMetadataHeader(this._securityMetadata);
+        const customerTrackingMetadataHeader = new CustomerTrackingMetadata(this._context.customerTrackingMetadata);
         this._interceptor = this._instance.interceptors.request.use(async request => {
             await this._authHeader.addAuthorizationHeader(this._memberId, request, this._context);
             versionHeader.addVersionHeader(request);
             developerHeader.addDeveloperHeader(request);
-            securityMetadataHeader.addSecurityMetadataHeader(request);
+            customerTrackingMetadataHeader.addCustomerTrackingData(request);
```
Comment (yuhanniu) [inline core/src/http/AuthHttpClient.js]: I am not sure if this is the right way to do it, the new headers are just like the exisiting `onBehalfOf` or `customerInitiated` headers, please refer to their implementations.

File: core/src/http/AuthHeader.js (hunk 2 above)
Comment (priyanshu-ap) [reply core/src/http/AuthHeader.js]: done

File: core/src/http/AuthHttpClient.js (hunk 3 above)
Comment (priyanshu-ap) [reply core/src/http/AuthHttpClient.js]: done

File: tpp/src/main/Member.js [hunk 4]
```diff
@@ -78,13 +79,21 @@ export default class Member extends CoreMember {
      * Creates a representable that acts as another member via an access token.
      *
      * @param accessTokenId - ID of the access token
+     * @param {CustomerTrackingMetadata} customerTrackingMetadata
      * @return new member that acts as another member
      */
-    forAccessToken(accessTokenId: string): Representable {
+    forAccessToken(accessTokenId: string, customerTrackingMetadata: CustomerTrackingMetadata): Representable {
```
Comment (yuhanniu) [inline tpp/src/main/Member.js]: Could you also add the customer initiated flag here? (mirroring the java sdk)
Basically the customer tracking metadata would only make sense if the request is initiated by the customer.

**Pull request merged**